modules/compliance-tests/src/smithy4s/compliancetests/internals/ClientHttpComplianceTestCase.scala

```diff
@@ -224,26 +224,12 @@ private[compliancetests] class ClientHttpComplianceTestCase[
   }
 
   def allClientTests(): List[ComplianceTest[F]] = {
-    service.endpoints.flatMap { case endpoint =>
-      val requestTests = endpoint.hints
-        .get(HttpRequestTests)
-        .map(_.value)
-        .getOrElse(Nil)
-        .filter(_.protocol == protocolTag.id.toString())
-        .filter(tc => tc.appliesTo.forall(_ == AppliesTo.CLIENT))
-        .map(tc => clientRequestTest(endpoint, tc))
-
-      val opResponseTests = endpoint.hints
-        .get(HttpResponseTests)
-        .map(_.value)
-        .getOrElse(Nil)
-        .filter(_.protocol == protocolTag.id.toString())
-        .filter(tc => tc.appliesTo.forall(_ == AppliesTo.CLIENT))
-        .map(tc => clientResponseTest(endpoint, tc))
-
-      val errorResponseTests = endpoint.errorable.toList
-        .flatMap { errorrable =>
-          errorrable.error.alternatives.flatMap { errorAlt =>
```

Comment:
essentially, I moved all of the logic inside that scope into  `toResponse` where the `endpoint` var is all typed up properly

Reply:
> I won't approve because mergify will go ahead and merge and the bump to scala 3.3.0 brings in a lot more changes that I had anticipated.
> 
> I'll let @kubukoz or @Baccata do that

Marked as draft in the hope that the mergify bot won't approve and merge it on CI success automatically